src/cryptoadvance/specter/specter.py

```diff
@@ -100,26 +102,36 @@ def check(self):
             self._info["chain"] = None
 
         chain = self._info["chain"]
-        if self.device_manager is None:
-            self.device_manager = DeviceManager(os.path.join(self.data_folder, "devices"))
+        if hasattr(user, 'is_admin'):
+            user_folder_id = '_' + user.id if user and not user.is_admin else ''
         else:
-            self.device_manager.update()
-
-        if self.wallet_manager is None or chain is None:
-            wallets_path = "specter%s" % self.config["uid"]
-            self.wallet_manager = WalletManager(
-                os.path.join(self.data_folder, "wallets"), 
-                self.cli, 
-                chain,
-                self.device_manager,
-                path=wallets_path
-            )
-        else:
-            self.wallet_manager.update(
-                os.path.join(self.data_folder, "wallets"), 
-                self.cli, 
-                chain=chain
-            )
+            user_folder_id = ''
+
+        if self.config['auth'] != 'usernamepassword' or (user and not user.is_anonymous):
+            if self.device_manager is None:
+                self.device_manager = DeviceManager(os.path.join(self.data_folder, "devices{}".format(user_folder_id)))
+            else:
+                self.device_manager.update()
```

Comment:
Probably you want to pass user path here as well, similar to wallet manager.

Reply:
Added. For me it works well, can you please verify now there's no persistence issue for you too?

Reply:
Works now!